app/models/user.rb

```diff
@@ -175,7 +175,7 @@ def active_mhv_ids
   end
 
   def address
-    address = identity&.address || mpi_profile&.address
+    address = identity&.address || mpi_profile&.address || {}
```

Comment:
Can you wrap this in an `OpenStruct` and then refer to the fields as functions? As in:

`address = OpenStruct.new(identity&.address || mpi_profile&.address)`
`{ street: address.street, street2: address.street2 ...`

I think that more closely will represent what we intended here, as in, to always return a hash with fields, even if they are empty (this is how the `identity.address` or `mpi_profile.address` functions return their address objects). It should also solve the case when the identity address object or MPI address object was somehow never defined in the first place, which is what you're trying to accomplish here

Reply:
Actually I suppose an empty hash accomplishes the same thing, for some reason I was thinking it was returning `nil`, instead this is functionally doing exactly what I asked above, haha.

In that case either implementation should be safe, it was unintentional that calling `user.address` would return an error, even if address wasn't defined at all

Reply:
Yea that is what I was referring to in my comment 
>Although there is another method Ive seen used that does what you want already that might be better than the additional "or"

I hadn't had time to dig that up. Thanks!!

Reply:
Actually, yea, you are right, this is fine I think as is.